### PR TITLE
Add additional options added to os_networks since 2.8

### LINKED ISSUE #17: Ability to define alternative MTU for networks
It would be useful to be able to set the MTU on networks that have jumbo frames (or indeed frames of any non-default size).  This is supported in the Ansible module here: https://docs.ansible.com/ansible/latest/modules/os_network_module.html?highlight=mtu

### PR DESCRIPTION
**Pull request by jovial**
This allows you to set the MTU. Fixes #17.

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -39,6 +39,9 @@
     external: "{{ item.external | default(omit) }}"
     project: "{{ item.project | default(omit) }}"
     state: "{{ item.state | default(omit) }}"
+    mtu: "{{ item.mtu | default(omit) }}"
```
Comment (markgoddard) [inline tasks/main.yml:42]: I think if you try to run this on Ansible 2.8 it will fail. That was behaviour previously at least

Comment (oneswig) [reply tasks/main.yml:42]: It claims to have been added in 2.9, according to https://docs.ansible.com/ansible/latest/modules/os_network_module.html - however if I try it on a production system I get:

`BadRequestException: 400: Client Error for url: https://......:9696/v2.0/networks.json, Invalid input for operation: Requested MTU is too big, maximum is 1500.`

... although setting the MTU to 9000 on the CLI appears to work fine.

**Review by oneswig — COMMENTED** [review]
Should we also bump the required Ansible version for this release of os_networks?

**Comment by jovial** [comment]
> Should we also bump the required Ansible version for this release of os_networks?

We could use argument splatting to maintain compatibility with older versions:

https://github.com/ansible/ansible/pull/43798

This would allow us to support >= 2.7, thoughts?

**Comment by jovial** [comment]
From my limited testing it seems you can pass arguments with omit with ansible>=2.5:
```
(t6) [will@juno ~]$ cat test5.yml 
---
- hosts: localhost
  tasks:
    - copy:
        src: /tmp/test.yml
        dest: /tmp/sfsfsad.yml
        madeupparam: "{{ hello | default(omit) }}"
```

```

(t6) [will@juno ~]$ pip install -U --force 'ansible<2.6'
Processing ./.cache/pip/wheels/5d/e4/6c/4dbd88e9fc05c66792127952b466288b18f1c7acc6a1e8a73a/ansible-2.5.15-cp38-none-any.whl
Processing ./.cache/pip/wheels/d9/45/dd/65f0b38450c47cf7e5312883deb97d065e030c5cca0a365030/PyYAML-5.1.2-cp38-cp38-linux_x86_64.whl
Collecting cryptography
  Using cached https://files.pythonhosted.org/packages/ca/9a/7cece52c46546e214e10811b36b2da52ce1ea7fa203203a629b8dfadad53/cryptography-2.8-cp34-abi3-manylinux2010_x86_64.whl
Collecting setuptools
  Using cached https://files.pythonhosted.org/packages/9e/d5/444a443d890f09fc1ca1a2c3c9fc7e84cb148177b05ac94fe5084e3d9abb/setuptools-42.0.1-py2.py3-none-any.whl
Collecting jinja2
  Using cached https://files.pythonhosted.org/packages/65/e0/eb35e762802015cab1ccee04e8a277b03f1d8e53da3ec3106882ec42558b/Jinja2-2.10.3-py2.py3-none-any.whl
Collecting paramiko
  Using cached https://files.pythonhosted.org/packages/4b/80/74dace9e48b0ef923633dfb5e48798f58a168e4734bca8ecfaf839ba051a/paramiko-2.6.0-py2.py3-none-any.whl
Collecting six>=1.4.1
  Using cached https://files.pythonhosted.org/packages/65/26/32b8464df2a97e6dd1b656ed26b2c194606c16fe163c695a992b36c11cdf/six-1.13.0-py2.py3-none-any.whl
Collecting cffi!=1.11.3,>=1.8
  Using cached https://files.pythonhosted.org/packages/05/7d/723ceca757d8016850d1110df44b91121695f22f782f876b483ecb3928a9/cffi-1.13.2-cp38-cp38-manylinux1_x86_64.whl
Processing ./.cache/pip/wheels/f2/aa/04/0edf07a1b8a5f5f1aed7580fffb69ce8972edc16a505916a77/MarkupSafe-1.1.1-cp38-cp38-linux_x86_64.whl
Collecting bcrypt>=3.1.3
  Using cached https://files.pythonhosted.org/packages/8b/1d/82826443777dd4a624e38a08957b975e75df859b381ae302cfd7a30783ed/bcrypt-3.1.7-cp34-abi3-manylinux1_x86_64.whl
Collecting pynacl>=1.0.1
  Using cached https://files.pythonhosted.org/packages/27/15/2cd0a203f318c2240b42cd9dd13c931ddd61067809fee3479f44f086103e/PyNaCl-1.3.0-cp34-abi3-manylinux1_x86_64.whl
Processing ./.cache/pip/wheels/f2/9a/90/de94f8556265ddc9d9c8b271b0f63e57b26fb1d67a45564511/pycparser-2.19-py2.py3-none-any.whl
Installing collected packages: PyYAML, six, pycparser, cffi, cryptography, setuptools, MarkupSafe, jinja2, bcrypt, pynacl, paramiko, ansible
  Found existing installation: PyYAML 5.1.2
    Uninstalling PyYAML-5.1.2:
      Successfully uninstalled PyYAML-5.1.2
  Found existing installation: six 1.13.0
    Uninstalling six-1.13.0:
      Successfully uninstalled six-1.13.0
  Found existing installation: pycparser 2.19
    Uninstalling pycparser-2.19:
      Successfully uninstalled pycparser-2.19
  Found existing installation: cffi 1.13.2
    Uninstalling cffi-1.13.2:
      Successfully uninstalled cffi-1.13.2
  Found existing installation: cryptography 2.8
    Uninstalling cryptography-2.8:
      Successfully uninstalled cryptography-2.8
  Found existing installation: setuptools 42.0.1
    Uninstalling setuptools-42.0.1:
      Successfully uninstalled setuptools-42.0.1
  Found existing installation: MarkupSafe 1.1.1
    Uninstalling MarkupSafe-1.1.1:
      Successfully uninstalled MarkupSafe-1.1.1
  Found existing installation: Jinja2 2.10.3
    Uninstalling Jinja2-2.10.3:
      Successfully uninstalled Jinja2-2.10.3
  Found existing installation: bcrypt 3.1.7
    Uninstalling bcrypt-3.1.7:
      Successfully uninstalled bcrypt-3.1.7
  Found existing installation: PyNaCl 1.3.0
    Uninstalling PyNaCl-1.3.0:
      Successfully uninstalled PyNaCl-1.3.0
  Found existing installation: paramiko 2.6.0
    Uninstalling paramiko-2.6.0:
      Successfully uninstalled paramiko-2.6.0
  Found existing installation: ansible 2.5.15
    Uninstalling ansible-2.5.15:
      Successfully uninstalled ansible-2.5.15
Successfully installed MarkupSafe-1.1.1 PyYAML-5.1.2 ansible-2.5.15 bcrypt-3.1.7 cffi-1.13.2 cryptography-2.8 jinja2-2.10.3 paramiko-2.6.0 pycparser-2.19 pynacl-1.3.0 setuptools-42.0.1 six-1.13.0
(t6) [will@juno ~]$ ansible-playbook test5.yml
 [WARNING]: Unable to parse /etc/ansible/hosts as an inventory source

 [WARNING]: No inventory was parsed, only implicit localhost is available

 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'


PLAY [localhost] ************************************************************************************************************************************************************

TASK [Gathering Facts] ******************************************************************************************************************************************************
ok: [localhost]

TASK [copy] *****************************************************************************************************************************************************************
ok: [localhost]

PLAY RECAP ******************************************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0

(t6) [will@juno ~]$ pip install -U --force 'ansible<2.5'
Processing ./.cache/pip/wheels/db/8e/a9/1aa2d1ed37b34b4f07fa3860e2adeb6bf0d943dd38cd07c928/ansible-2.4.6.0-cp38-none-any.whl
Collecting jinja2
  Using cached https://files.pythonhosted.org/packages/65/e0/eb35e762802015cab1ccee04e8a277b03f1d8e53da3ec3106882ec42558b/Jinja2-2.10.3-py2.py3-none-any.whl
Collecting paramiko
  Using cached https://files.pythonhosted.org/packages/4b/80/74dace9e48b0ef923633dfb5e48798f58a168e4734bca8ecfaf839ba051a/paramiko-2.6.0-py2.py3-none-any.whl
Collecting setuptools
  Using cached https://files.pythonhosted.org/packages/9e/d5/444a443d890f09fc1ca1a2c3c9fc7e84cb148177b05ac94fe5084e3d9abb/setuptools-42.0.1-py2.py3-none-any.whl
Processing ./.cache/pip/wheels/d9/45/dd/65f0b38450c47cf7e5312883deb97d065e030c5cca0a365030/PyYAML-5.1.2-cp38-cp38-linux_x86_64.whl
Collecting cryptography
  Using cached https://files.pythonhosted.org/packages/ca/9a/7cece52c46546e214e10811b36b2da52ce1ea7fa203203a629b8dfadad53/cryptography-2.8-cp34-abi3-manylinux2010_x86_64.whl
Processing ./.cache/pip/wheels/f2/aa/04/0edf07a1b8a5f5f1aed7580fffb69ce8972edc16a505916a77/MarkupSafe-1.1.1-cp38-cp38-linux_x86_64.whl
Collecting bcrypt>=3.1.3
  Using cached https://files.pythonhosted.org/packages/8b/1d/82826443777dd4a624e38a08957b975e75df859b381ae302cfd7a30783ed/bcrypt-3.1.7-cp34-abi3-manylinux1_x86_64.whl
Collecting pynacl>=1.0.1
  Using cached https://files.pythonhosted.org/packages/27/15/2cd0a203f318c2240b42cd9dd13c931ddd61067809fee3479f44f086103e/PyNaCl-1.3.0-cp34-abi3-manylinux1_x86_64.whl
Collecting six>=1.4.1
  Using cached https://files.pythonhosted.org/packages/65/26/32b8464df2a97e6dd1b656ed26b2c194606c16fe163c695a992b36c11cdf/six-1.13.0-py2.py3-none-any.whl
Collecting cffi!=1.11.3,>=1.8
  Using cached https://files.pythonhosted.org/packages/05/7d/723ceca757d8016850d1110df44b91121695f22f782f876b483ecb3928a9/cffi-1.13.2-cp38-cp38-manylinux1_x86_64.whl
Processing ./.cache/pip/wheels/f2/9a/90/de94f8556265ddc9d9c8b271b0f63e57b26fb1d67a45564511/pycparser-2.19-py2.py3-none-any.whl
Installing collected packages: MarkupSafe, jinja2, six, pycparser, cffi, cryptography, bcrypt, pynacl, paramiko, setuptools, PyYAML, ansible
  Found existing installation: MarkupSafe 1.1.1
    Uninstalling MarkupSafe-1.1.1:
      Successfully uninstalled MarkupSafe-1.1.1
  Found existing installation: Jinja2 2.10.3
    Uninstalling Jinja2-2.10.3:
      Successfully uninstalled Jinja2-2.10.3
  Found existing installation: six 1.13.0
    Uninstalling six-1.13.0:
      Successfully uninstalled six-1.13.0
  Found existing installation: pycparser 2.19
    Uninstalling pycparser-2.19:
      Successfully uninstalled pycparser-2.19
  Found existing installation: cffi 1.13.2
    Uninstalling cffi-1.13.2:
      Successfully uninstalled cffi-1.13.2
  Found existing installation: cryptography 2.8
    Uninstalling cryptography-2.8:
      Successfully uninstalled cryptography-2.8
  Found existing installation: bcrypt 3.1.7
    Uninstalling bcrypt-3.1.7:
      Successfully uninstalled bcrypt-3.1.7
  Found existing installation: PyNaCl 1.3.0
    Uninstalling PyNaCl-1.3.0:
      Successfully uninstalled PyNaCl-1.3.0
  Found existing installation: paramiko 2.6.0
    Uninstalling paramiko-2.6.0:
      Successfully uninstalled paramiko-2.6.0
  Found existing installation: setuptools 42.0.1
    Uninstalling setuptools-42.0.1:
      Successfully uninstalled setuptools-42.0.1
  Found existing installation: PyYAML 5.1.2
    Uninstalling PyYAML-5.1.2:
      Successfully uninstalled PyYAML-5.1.2
  Found existing installation: ansible 2.5.15
    Uninstalling ansible-2.5.15:
      Successfully uninstalled ansible-2.5.15
Successfully installed MarkupSafe-1.1.1 PyYAML-5.1.2 ansible-2.4.6.0 bcrypt-3.1.7 cffi-1.13.2 cryptography-2.8 jinja2-2.10.3 paramiko-2.6.0 pycparser-2.19 pynacl-1.3.0 setuptools-42.0.1 six-1.13.0
(t6) [will@juno ~]$ ansible-playbook test5.yml
 [WARNING]: Unable to parse /etc/ansible/hosts as an inventory source

 [WARNING]: No inventory was parsed, only implicit localhost is available

 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'


PLAY [localhost] ************************************************************************************************************************************************************
ERROR! Unexpected Exception, this is probably a bug: invalid syntax (task_executor.py, line 568)
to see the full traceback, use -vvv
```
So it would seem this would work in a backwards compatible way. Maybe I'm missing something?

**Review by markgoddard — APPROVED** [review]

**Pull request merged**